sdk/dnsresolver/Microsoft.Azure.Management.DnsResolver/tests/Extensions/ResourceManagementClientExtensions.cs

```diff
@@ -0,0 +1,67 @@
+// ------------------------------------------------------------------------------------------------
+// <copyright file="ResourceManagementClientExtensions.cs" company="Microsoft Corporation">
+//   Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+// ------------------------------------------------------------------------------------------------
+
+namespace DnsResolver.Tests.Extensions
+{
+    using System;
+    using System.Linq;
+    using Microsoft.Azure.Management.Resources;
+    using Microsoft.Azure.Management.Resources.Models;
+
+    internal static class ResourceManagementClientExtensions
+    {
+        public static ResourceGroup CreateResourceGroup(
+            this ResourceManagementClient client,
+            string resourceGroupName = null,
+            string resourceGroupLocation = null)
+        {
+            if (client == null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
+
+            resourceGroupName = resourceGroupName ?? TestDataGenerator.GenerateResourceGroupName();
+            var response = client.ResourceGroups.CreateOrUpdate(resourceGroupName, TestDataGenerator.GenerateResourceGroup(resourceGroupLocation));
+
+            return response;
+        }
+
+        public static string GetResourceLocation(this ResourceManagementClient client, string resourceType)
+        {
+            if (client == null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
+
+            string location = null;
+
+            var parts = resourceType.Split('/');
+            var providerName = parts[0];
+            var resourceName = parts[1];
+
+            var provider = client.Providers.Get(providerName);
+            foreach (var resource in provider.ResourceTypes)
+            {
+                if (string.Equals(resource.ResourceType, resourceName, StringComparison.OrdinalIgnoreCase))
+                {
+                    location = resource.Locations.FirstOrDefault(loc => !string.IsNullOrEmpty(loc));
+                }
+            }
+
+            return location;
+        }
+
+        public static string GetDnsResolverResourceLocation(this ResourceManagementClient client)
+        {
+            if (client == null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
+
+            return client.GetResourceLocation("Microsoft.Network/dnsresolvers");
```

Comment:
Let's no use hardcodes anywhere in code. We should have a common class with strings such as this.